rasa/nlu/featurizers/featurizer.py

```diff
@@ -12,7 +12,10 @@ def __init__(self, component_config: Optional[Dict[Text, Any]] = None) -> None:
             component_config = {}
 
         # makes sure the alias name is set
-        component_config.setdefault(FEATURIZER_CLASS_ALIAS, self.name)
+        self.component_config = (
+            component_config  # Necessary for `unique_name` to be defined
+        )
```

Comment:
I think this would be a bit more elegant
```suggestion
        # Necessary for `unique_name` to be defined
        self.component_config = component_config
```